servers/src/main/java/tachyon/worker/block/evictor/LRFUEvictor.java

```diff
@@ -0,0 +1,345 @@
+import tachyon.exception.NotFoundException;
+import tachyon.worker.block.BlockMetadataManagerView;
+import tachyon.worker.block.BlockStoreEventListenerBase;
+import tachyon.worker.block.BlockStoreLocation;
+import tachyon.worker.block.meta.BlockMeta;
+import tachyon.worker.block.meta.StorageDir;
+import tachyon.worker.block.meta.StorageDirView;
+import tachyon.worker.block.meta.StorageTier;
+import tachyon.worker.block.meta.StorageTierView;
+
+/**
+ * This class is used to evict blocks by LRFU. LRFU combines LRU and LFU, it evicts blocks with
+ * small frequency or large recency. Actually, LRFU evicts blocks with minimum CRF. CRF of a block
+ * is the sum of F(t) = pow((1.0 / {@link #mAttenuationFactor}, t * {@link #mStepFactor}) and each
+ * access to a block has a F(t) value where t is the time interval since that access time to 
+ * current. When {@link #mStepFactor} is close to 0, LRFU is close to LFU. Conversely, LRFU is 
+ * close to LRU when {@link #mStepFactor} is close to 1
+ */
+public class LRFUEvictor extends BlockStoreEventListenerBase implements Evictor {
+
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
+  private static final double DEFAULT_STEP_FACTOR = 0.25;
+  private static final double DEFAULT_ATTENUATION_FACTOR = 2.0;
+
+  private BlockMetadataManagerView mManagerView;
+  // Map from block id to the last access time of the block
+  private Map<Long, Long> mBlockIdToLastAccessTime = new ConcurrentHashMap<Long, Long>();
+  // Map from block id to the CRF value of the block
+  private Map<Long, Double> mBlockIdToCRFValue = new ConcurrentHashMap<Long, Double>();
+  // In the range of [0, 1]. Closer to 0, LRFU closer to LFU. Closer to 1, LRFU closer to LRU
+  private final double mStepFactor;
+  // In the range of [2, INF]
+  private final double mAttenuationFactor;
+  private final TachyonConf mTachyonConf;
+
+  //logic time
+  private AtomicLong mLogicTimeCount = new AtomicLong(0L);
+
+  public LRFUEvictor(BlockMetadataManagerView view) {
+    mManagerView = view;
+    mTachyonConf = new TachyonConf();
+    mStepFactor = mTachyonConf
+        .getDouble(Constants.WORKER_EVICT_STRATEGY_LRFU_STEP_FACTOR, DEFAULT_STEP_FACTOR);
+    mAttenuationFactor = mTachyonConf
+        .getDouble(Constants.WORKER_EVICT_STRATEGY_LRFU_ATTENUATION_FACTOR, 
+            DEFAULT_ATTENUATION_FACTOR);
+
+    // Preloading blocks
+    for (StorageTierView tier : mManagerView.getTierViews()) {
+      for (StorageDirView dir : tier.getDirViews()) {
+        for (BlockMeta block : dir.getEvictableBlocks()) {
+          mBlockIdToLastAccessTime.put(block.getBlockId(), 0L);
+          mBlockIdToCRFValue.put(block.getBlockId(), 0.0);
+        }
+      }
+    }
+  }
+
+  /**
+   * @return a StorageDirView in the range of location that already has availableBytes larger than
+   *         bytesToBeAvailable, otherwise null
+   */
+  private StorageDirView selectDirWithRequestedSpace(long bytesToBeAvailable,
```

Comment:
Since this method is identical (is it?) to `LRUEvictor.selectDirWithRequestedSpace`, could you leave a TODO here to make it maybe into `EvictorUtils`?